control/controlclient/direct.go

```diff
@@ -1417,10 +1414,19 @@ func doPingerPing(logf logger.Logf, c *http.Client, pr *tailcfg.PingRequest, pin
 		return
 	}
 	start := time.Now()
-	pinger.Ping(pr.IP, pingType, func(res *ipnstate.PingResult) {
-		// Currently does not check for error since we just return if it fails.
-		postPingResult(start, logf, c, pr, res.ToPingResponse(pingType))
-	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	res, err := pinger.Ping(ctx, pr.IP, pingType)
+	if err != nil {
+		if ctx.Err() != nil {
+			return
+		}
```

Comment:
that means that you won't log context timeout errors, why not?

Reply:
Also maybe log a ping request so that we can confirm that it made it to the client?